### PR TITLE
Modernize the Vault Genesis Kit

### PR DESCRIPTION
**Pull request by jhunt**
This commit rips out the kit.yml way of doing things in favor of newer
Genesis v2.4+ hooks like `blueprint` and `new`.  It also introduces the
use of credhub, colocated on the deploying BOSH director, for
implementation certificates (like consul)

This has been vetted manually in an AWS environment.

### CODE REVIEW

**Comment by Proplex** [comment]
LGTM—tested on GCP and vSphere, both worked as expected. Documentation looks good as well.

**Comment by patrick1378** [comment]
No issues from a user perspective from deploying on both GCP and vsphere 👌 .

File: MANUAL.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+```
+params:
+  vault_network:   vault
+  vault_disk_pool: vault # should be at least 1GB
+  vault_vm_type:   small # VMs should have at least 1 CPU, and 1GB of memory
+```
+
+# Available Features
+
+- `azure` - Replaces BOSH's native "availability zones"
```
Comment (dennisjbell) [inline MANUAL.md]: Azure isn't a feature anymore, the bosh_cpi is checked directly using the blueprint hook.

File: MANUAL.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  and `z3`.
+
+# Cloud Configuration
+
+By default, Vault uses the following VM types/networks/disk pools from your
+cloud config. Feel free to override them in your environment, if you would
+rather they use entities already existing in your cloud config:
+
+```
+params:
```
Comment (dennisjbell) [inline MANUAL.md]: These don't match the actual:
```
  vault_disk_pool: default # should be at least 1GB
  vault_vm_type:   default # VMs should have at least 1 CPU, and 1GB of memory
```

File: MANUAL.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+credentials used elsewhere in deployments and applications.
+
+# Base Parameters
+
+- `vault_disk_pool` - The persistent disk pool that Vault VMs will
+  use.  This pool must exist in your cloud config.  Defaults to
+  `vault`.
+
+- `vault_vm_type` - What type of VM to deploy.  This type must
+  exist in your cloud config.  Defaults to `small`.
```
Comment (dennisjbell) [inline MANUAL.md]: Default is actually `default`

File: MANUAL.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+The **Vault Genesis Kit** deploys a consul-backed Vault cluster.
+Vault provides secure storage of and access to sensitive
+credentials used elsewhere in deployments and applications.
+
+# Base Parameters
+
+- `vault_disk_pool` - The persistent disk pool that Vault VMs will
+  use.  This pool must exist in your cloud config.  Defaults to
+  `vault`.
```
Comment (dennisjbell) [inline MANUAL.md]: Default is actually `default`

File: README.md [hunk 5]
```diff
@@ -24,62 +23,13 @@ genesis init --kit vault/1.0.0
 genesis init --kit vault -d my-vault-configs
```
Comment (dennisjbell) [inline README.md]: Do we really want to promote/acknowledge the fact that you can have differently named repos?

Comment (jhunt) [reply README.md]: That's just scaffolded from `genesis create-kit`.  You'd have to talk to the guy who manages that project.

Comment (dennisjbell) [reply README.md]: Its the can-we/should-we debate... at one time we definitely wanted to allow the user to do just about anything, but as we get closer to wrapping this all in a `codex` repo, just not sure how we want to proceed.

**Pull request merged**